test/controllers/PenaltyDetailsController.test.ts

```diff
@@ -82,8 +85,16 @@ describe('PenaltyDetailsController', () => {
                 }
             });
 
+            companyProfileService.getCompanyProfile('SC123123').resolves({
```

Comment:
It's great that we have a test for the happy path here, but what happens when this service returns an error?

Reply:
well spotted